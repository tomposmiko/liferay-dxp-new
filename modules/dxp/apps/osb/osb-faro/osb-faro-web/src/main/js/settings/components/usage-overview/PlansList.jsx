import Panel from 'shared/components/Panel';
import PlanBreakdown from './PlanBreakdown';
import React from 'react';
import SubscriptionTitle from './SubscriptionTitle';
import {
	getPlanAddOns,
	getPropLabel,
	INDIVIDUALS,
	PAGEVIEWS,
	PLAN_TYPES,
	PLANS
} from 'shared/util/subscriptions';
import {PropTypes} from 'prop-types';

class PlansList extends React.Component {
	static defaultProps = {
		plans: []
	};

	static propTypes = {
		currentPlanName: PropTypes.string,
		plans: PropTypes.arrayOf(
			PropTypes.shape({
				baseSubscriptionPlan: PropTypes.string,
				limits: PropTypes.shape({
					[INDIVIDUALS]: PropTypes.number,
					[PAGEVIEWS]: PropTypes.number
				}),
				name: PropTypes.string,
				price: PropTypes.number
			})
		)
	};

	render() {
		const {currentPlanName, plans} = this.props;

		return (
			<div
				className={`plans-list-root${
					this.props.className ? ` ${this.props.className}` : ''
				}`}
			>
				{plans.map(({limits, name, price}, index) => {
					const isCurrentPlan = currentPlanName === name;

					return (
						<Panel
							className={
								this.props.className
									? ` ${this.props.className}`
									: ''
							}
							expandable
							initialExpanded={index === 0}
							key={name}
							title={
								<SubscriptionTitle
									labelText={
										isCurrentPlan
											? Liferay.Language.get(
													'current-plan'
											  )
											: ''
									}
									name={getPropLabel(name)}
									price={!isCurrentPlan ? price : -1}
								/>
							}
						>
							<PlanBreakdown
								addOns={
									name === PLANS.basic.name
										? []
										: getPlanAddOns(PLAN_TYPES[name])
								}
								limits={[
									{
										entityLabel: INDIVIDUALS,
										value: limits.individuals
									},
									{
										entityLabel: PAGEVIEWS,
										value: limits.pageViews
									}
								]}
							/>
						</Panel>
					);
				})}
			</div>
		);
	}
}

export default PlansList;
