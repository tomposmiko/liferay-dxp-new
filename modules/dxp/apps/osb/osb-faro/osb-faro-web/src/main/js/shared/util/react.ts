import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
