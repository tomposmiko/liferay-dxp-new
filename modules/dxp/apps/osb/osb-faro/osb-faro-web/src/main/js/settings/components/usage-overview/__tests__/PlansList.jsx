import PlansList from '../PlansList';
import React from 'react';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

const mockPlans = [
	{
		baseSubscriptionPlan: null,
		limits: {
			individuals: 0,
			pageViews: 0
		},
		name: 'Liferay Analytics Cloud Business',
		price: 0
	},
	{
		baseSubscriptionPlan: null,
		limits: {
			individuals: 5,
			pageViews: 30
		},
		name: 'Liferay Limited Business',
		price: 45
	}
];

describe('PlansList', () => {
	it('should render', () => {
		const {container} = render(<PlansList />);

		expect(container).toMatchSnapshot();
	});

	it('should render multiple plans', () => {
		const {container} = render(<PlansList plans={mockPlans} />);

		expect(container.querySelectorAll('.plan-name').length).toBeGreaterThan(
			1
		);
	});

	it('should render with a label in the list of plans for the current plan', () => {
		const {getByText} = render(
			<PlansList
				currentPlanName='Liferay Analytics Cloud Business'
				plans={mockPlans}
			/>
		);

		expect(getByText('Current Plan')).toBeTruthy();
	});
});
