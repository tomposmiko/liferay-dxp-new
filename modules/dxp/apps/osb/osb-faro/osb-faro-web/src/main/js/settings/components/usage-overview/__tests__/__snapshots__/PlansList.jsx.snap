// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlansList should render 1`] = `
<div>
  <div
    class="plans-list-root"
  />
</div>
`;
